apps/web/scripts/kernel/handlers/common.ts

```diff
@@ -27,9 +31,13 @@ export const handleRequestPayload: HandleMethod = ({
 	responseMethod,
 }) => {
 	const { sourceRequestId, requestId } = payload;
-	const { payload: sourcePayload } = getRequestRecord(sourceRequestId);
+	const { payload: sourcePayload, channel: sourceChannel } =
+		getRequestRecord(sourceRequestId);
 
-	responseMethod(requestId, ResponseCode.SUCCESS, sourcePayload);
+	responseMethod(requestId, ResponseCode.SUCCESS, {
+		...sourceChannel,
```

Comment:
@FME849 can you explain why we need source channel in the response body?

Reply:
this is the response for `requestPayload` from popup, the source channel include a hidden field called `sender` that has info about dapp (i.e domain, url, title,...)